arangod/Indexes/Index.cpp

```diff
@@ -261,6 +261,20 @@ size_t Index::sortWeight(arangodb::aql::AstNode const* node) {
   }
 }
 
+/// @brief validate fields that start or end with ":"
+void Index::validateFieldsWithSpecialCase(VPackSlice const& fields) {
+  if (!fields.isArray()) {
+    return;
+  }
+  for (VPackSlice name : VPackArrayIterator(fields)) {
+    if (name.toString().starts_with(":") || name.toString().ends_with(":")) {
```

Comment:
Avoid creation of 2 `std::string` temporaries.
```suggestion
    if (name.stringView().starts_with(":") || name.stringView().ends_with(":")) {
```